quiche/src/recovery/mod.rs

```diff
@@ -1334,6 +1334,175 @@ mod tests {
         }
     }
 
+    // TODO: `congestion` and `gcongestion` behave differently. That might be ok
+    // given the different algorithms but it would be ideal to merge and share
+    // the logic.
+    #[rstest]
+    fn time_thresholds_on_reordering(
+        #[values("bbr2_gcongestion")] cc_algorithm_name: &str,
+    ) {
+        let mut cfg = Config::new(crate::PROTOCOL_VERSION).unwrap();
+        assert_eq!(cfg.set_cc_algorithm_name(cc_algorithm_name), Ok(()));
+
+        let mut r = Recovery::new(&cfg);
+
+        assert_eq!(r.rtt(), Duration::from_millis(333));
+        let initial_thresh = 333 as f64 * INITIAL_TIME_THRESHOLD;
+        let spurious_loss_thresh = 333 as f64 * PACKET_REORDER_TIME_THRESHOLD;
+        assert_eq!(initial_thresh.round(), 375.0);
+        assert_eq!(spurious_loss_thresh.round(), 416.0);
+
+        #[rustfmt::skip]
+        // Pick time between and above thresholds for testing threshold increase.
+        //
+        //```
+        // |--- INITIAL_PACKET_THRESHOLD --- PACKET_REORDER_TIME_THRESHOLD ------- |
+        //                                ^                                  ^
+        //                  (time_between_thresh)    (time_between_thresh + plus_overhead)
+        // ```
```

Comment:
I don't understand this comment.

Also, too wide - doesn't fit in my review window.